src/parser/lex.rs

```diff
@@ -1308,6 +1308,7 @@ fn unescape_string_literal(mut s: &str) -> String {
                 b'n' => '\n',
                 b'r' => '\r',
                 b't' => '\t',
+                b'\0' => '\0',
```

Comment:
```suggestion
                b'0' => '\0',
```
The character after the backslash is a literal `0` not null. This results in a panic (since it reaches the `unimplemented` below). The test you added was fine, but it only tests the lexer. The panic occurs in the parser (which runs after the lexer). I would suggest also adding a unit test to the `unescape_string_literal` function for the null character, and maybe consider adding it to our VRL test suite, in  a new file in `lib/tests/tests/expressions/literal` which will run it through the entirety of VRL.

I also want to point out you can use the VRL CLI (by running `cargo run` in `lib/cli`) when testing locally. This lets you easily test out new features like this.

Let me know if you need any help adding more tests for this. After this change, it appears to work for me in the CLI.